lib/li_box4d.c

```diff
@@ -0,0 +1,45 @@
+/***********************************************************************
+* li_box4d.c
+*
+***********************************************************************/
+
+#include "pc_api_internal.h"
+#include "li_api_internal.h"
+
+/**
+* Apply an affine transformation to a box4d.
+*/
+LIBOX4D *
+li_box4d_affine(
+	const LIBOX4D *ibox,
+	double a, double b, double c,
+	double d, double e, double f,
+	double g, double h, double i,
+	double xoff, double yoff, double zoff)
+{
+	LIMAT43 amat;
+	LIVEC3 vec, rvec;
+	LIBOX4D *obox = pcalloc(sizeof(LIBOX4D));
+
+	li_matrix_43_set(amat, a, b, c, xoff, d, e, f, yoff, g, h, i, zoff);
+
+	vec[0] = ibox->xmin;
+	vec[1] = ibox->ymin;
+	vec[2] = ibox->zmin;
+	li_matrix_43_transform_affine(rvec, amat, vec);
+	obox->xmin = rvec[0];
+	obox->ymin = rvec[1];
+	obox->zmin = rvec[2];
+	obox->mmin = ibox->mmin;
+
+	vec[0] = ibox->xmax;
+	vec[1] = ibox->ymax;
+	vec[2] = ibox->zmax;
+	li_matrix_43_transform_affine(rvec, amat, vec);
+	obox->xmax = rvec[0];
+	obox->ymax = rvec[1];
+	obox->zmax = rvec[2];
+	obox->mmax = ibox->mmax;
```

Comment:
There is a bug here. You need to take the bounding box of all 8 corners, rather than just 2.